### PR TITLE
Implement ability to use preemptible (spot) instances on GCE

### PR DESCRIPTION
**Pull request by enaydanov**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2449,6 +2481,7 @@ def __init__(self, cluster_uuid=None, cluster_prefix='cluster',
         self.log = SDCMAdapter(LOGGER, extra={'prefix': str(self)})
         self.log.info('Init nodes')
         self.nodes = []
+        self.instance_provision = params.get('instance_provision', default=INSTANCE_PROVISION_ON_DEMAND)
```
Comment (bentsi) [inline sdcm/cluster.py]: we set defaults for GCE backend in `defaults/gce_config.yaml`, please remove it  and
add `instance_provision: 'spot'` to the `defaults/gce_config.yaml`

Comment (enaydanov) [reply sdcm/cluster.py]: I'm in doubt about the removing a code from the base class. I've changed `defaults/gce_config.yaml`, but left `sdcm/cluster.py` unchanged. Please, confirm that I understand you correctly and I need to remove the initialization of `instance_provision` from the base class.

Comment (bentsi) [reply sdcm/cluster.py]: no, just remove  default from `params.get('instance_provision', default=INSTANCE_PROVISION_ON_DEMAND)`

Comment (bentsi) [reply sdcm/cluster.py]: (our `params` defaults are stored in `defaults` dir)

Comment (enaydanov) [reply sdcm/cluster.py]: Got it.

**Comment by bentsi** [comment]
@roydahan no, we needed to add `instance_provision: "spot"` to the `defaults/gce_config.yaml`
I will rerun it

**Comment by bentsi** [comment]
there are some issues with PR provision pipeline that prevent us from running it.
I will cherry-pick Evgeniy's commits and create a new PR

**Pull request closed without merging**